### ISSUE TITLE
mob with 0.22 broken for my setup

### ISSUE THREAD

**Issue by Morl99** [remotemobprogramming/mob#56]
Hello Simon,

while 0.20 worked really well, after the update to 0.22 using `mob start` and `mob next` now fails with this error message:
```
 ERROR git --no-pager log feature..mob/feature/removeDeployersYaml-DPL-2833 --pretty=format:%an --abbrev-commit
 ERROR fatal: ambiguous argument 'feature..mob/feature/removeDeployersYaml-DPL-2833': unknown revision or path not in the working tree.
 ERROR Use '--' to separate paths from revisions, like this:
 ERROR 'git <command> [<revision>...] -- [<file>...]'
 ERROR exit status 128

```

I have attached a full debug log of `mob start --branch ""`. If I understand the latest changes correctly, I always have to specify the --branch if I am not on master. But maybe I am misunderstanding. If I omit it, there is an error message telling me, I have to add `-- branch`.

[scratch_39.txt](https://github.com/remotemobprogramming/mob/files/4812659/scratch_39.txt)

**Comment by simonharrer** [comment]
You only need --branch when you want to have multiple wip branches of one base branch. And the other way round: if the tool detects a wip branch that has been created using the --branch parameter, it will automatically require you to add --branch.

**Comment by simonharrer** [comment]
Okay, I can reproduce your issue. Your branch name is 'feature/removeDeployersYaml-DPL-2833' . 'mob' currently doesn't work with branch names having the character '/' in them. The tool determines that 'feature' is the base branch and 'mob/feature/removeDeployersYaml-DPL-2833' is the wip branch with --branch value 'removeDeployersYaml-DPL-2833'. But that's not what you have, because there's no 'feature' branch, only 'feature/removeDeployersYaml-DPL-2833'. 

In other words, here's what mob thinks you came from: `feature $ mob start --branch removeDeployersYaml-DPL-2833 # switching to mob/feature/removeDeployersYaml-DPL-2833`.

I try to fix this by taking into account the available branches.

**Comment by simonharrer** [comment]
I think I might have a solution. Can you try it out? You must install 'mob' manually through cloning this repo and running `./install` with go installed.

**Comment by Morl99** [comment]
Tested it. The log now states:
```
   git fetch origin --prune
   git pull --ff-only
 > starting new mob session from origin/feature
 ERROR git checkout -B mob/feature/removeDeployersYaml-DPL-2833 origin/feature
 ERROR fatal: 'origin/feature' is not a commit and a branch 'mob/feature/removeDeployersYaml-DPL-2833' cannot be created from it
 ERROR exit status 128
```

I had manually cleaned up the mob branch on remote an local before trying the mob start. I think you are halfway there though.

Really appreciate the quick turnaround.

**Comment by simonharrer** [comment]
Can you provide the trace using the —debug flag? So I can try to narrow it down.

**Comment by Morl99** [comment]
Sorry, forgot about --debug
see attached file, but today the message is different again. I am not sure, what the setup was, that lead to the message from yesterday, we are now back to square one. I am relatively sure, that I am on #40f9b03cf4a8628eb7cb8878f726680ca06731e8 and did a fresh install. `mob version` shows `v0.0.23`.

Maybe you can reopen this issue until we get it fixed?

**Comment by simonharrer** [comment]
Can you try again with the latest commit on master?

**Comment by simonharrer** [comment]
I take the available branches into account. With them, I can determine the specifics.

**Comment by Morl99** [comment]
Awesome, now it works as expected :) Can you create a release with this fix?

**Comment by simonharrer** [comment]
Will release 0.0.24 right now.
